dkg/src/agg_trx_producer.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright © Aptos Foundation
+
+use crate::{
+    transcript_aggregation::TranscriptAggregationState, types::DKGNodeRequest, DKGMessage,
+};
+use aptos_channels::aptos_channel::Sender;
+use aptos_reliable_broadcast::ReliableBroadcast;
+use aptos_types::{dkg::DKGTrait, epoch_state::EpochState};
+use futures::future::AbortHandle;
+use futures_util::future::Abortable;
+use std::sync::Arc;
+use tokio_retry::strategy::ExponentialBackoff;
+
+/// A sub-process of the whole DKG process.
+/// Once invoked by `DKGManager` to `start_produce`,
+/// it starts producing an aggregated transcript and returns an abort handle.
+/// Once an aggregated transcript is available, it is sent back via channel `agg_trx_tx`.
+pub trait AggTranscriptProducer<S: DKGTrait>: Send + Sync {
```

Comment:
why this needs to be trait? for unit test?

Reply:
yes